File: Day5/Java/Main.java
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Main {

	public static void main(String[] args) throws IOException {
		File input = new File("input").getAbsoluteFile();
		if (!input.exists()) {
			System.out.println(input.getPath() + " does not exist!");
			System.out.println("trying " + input.getPath() + ".txt instead.");
			input = new File(input.getPath() + ".txt");
			if (!input.exists()) {
				System.err.println("neither " + input.getPath().substring(0, input.getPath().length() - 4) + ",");
				System.err.println("nor " + input.getPath() + " exists!");
				return;
			}
		}

		List<Integer> found = new ArrayList<>();
		for (String line : Files.readAllLines(input.toPath())) {
			line = line.replaceAll("[FL]", "0").replaceAll("[RB]", "1");

			found.add(Integer.parseInt(line, 2));
		}

		Collections.sort(found);
		System.out.println("Highest found seat id is " + found.get(found.size() - 1) + ".");

		int last = 0;
		for (int seat : found) {
			if (seat == last + 2) {
				last = seat - 1;
				break;
			}
			last = seat;
		}

		System.out.println("Your seat has id " + last + ".");
	}
}

File: Day5/Python/Main.py
import os
import re

def main():
    input = os.path.dirname(os.getcwd())
    input = os.path.join(input, "input")
    if not os.path.exists(input):
        print(input + " does not exist!")
        input = os.path.join(os.path.dirname(input), "input.txt")
        print(f"trying {input} instead.")
        if not os.path.exists(input):
            print(input + " does not exist!")
            return;

    inputFile = open(input, 'r')
    found = []
    for line in inputFile.readlines():
        line = re.sub("[FL]", "0", re.sub("[BR]", "1", line))

        found.append(int(line[:7], 2) * 8 + int(line[-4:], 2))

    found.sort()
    print(f"Highest found seat id is {found[-1]}.")

    last = 0
    for seat in found:
        if seat == last + 2:
            last = seat - 1
            break
        last = seat

    print(f"Your seat has id {last}.")

if __name__ == '__main__':
    main()
